### PR TITLE
Build opentelemetry-cpp if not present

### PR DESCRIPTION
**Pull request by lalitb**
Build opentelemetry-cpp from the repo if it's not already installed. 
Changes are confined to `exporters/fluentd/cmake/opentelemetry-cpp.cmake`, which is called from `exporters/fluentd/CMakeLists.txt`. There are a few formatting changes by cmake-format.

### CODE REVIEW

File: exporters/fluentd/cmake/opentelemetry-cpp.cmake [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    target_create("http_client_curl"
+                  "ext/src/http/client/curl/libhttp_client_curl.a")
+    add_library(opentelemetry::libopentelemetry INTERFACE IMPORTED)
+    add_dependencies(opentelemetry::libopentelemetry opentelemetry-cpp)
+    set_target_properties(
+        opentelemetry::libopentelemetry
+        PROPERTIES
+        INTERFACE_LINK_LIBRARIES "${opentelemetry_deps}")
+endfunction()
+
```
Comment (ThomsonTan) [inline exporters/fluentd/cmake/opentelemetry-cpp.cmake:75]: nit: remove the empty lines.

File: exporters/fluentd/CMakeLists.txt [hunk 2]
```diff
@@ -18,7 +18,20 @@ project(opentelemetry-fluentd)
...
+set(opentelemetry_cpp_clone FALSE)
+if(opentelemetry_cpp_FOUND)
+  message("Using external opentelemetry-cpp")
+else()
+  include(cmake/opentelemetry-cpp.cmake)
+  build_opentelemetry()
+  set(opentelemetry_cpp_clone TRUE)
+  set(OPENTELEMETRY_CPP_INCLUDE_DIRS "")
+  set(OPENTELEMETRY_CPP_LIBRARIES "opentelemetry::libopentelemetry")
+  message("\nopentelemetry-cpp package was not found. Cloning from github")
```
Comment (ThomsonTan) [inline exporters/fluentd/CMakeLists.txt:32]:
```suggestion
  message("\nopentelemetry-cpp package was not found. Cloned from github")
```

**Review by ThomsonTan — APPROVED** [review]

File: exporters/fluentd/CMakeLists.txt [hunk 3]
```diff
@@ -18,7 +18,20 @@ project(opentelemetry-fluentd)
 
 add_definitions(-DHAVE_CONSOLE_LOG)
 add_definitions(-DENABLE_LOGS_PREVIEW)
-find_package(opentelemetry-cpp REQUIRED)
+
+find_package(opentelemetry-cpp QUIET)
+set(opentelemetry_cpp_clone FALSE)
```
Comment (ThomsonTan) [inline exporters/fluentd/CMakeLists.txt:23]: Is this variable used?

File: exporters/fluentd/cmake/opentelemetry-cpp.cmake (hunk 1 above)
Comment (lalitb) [reply exporters/fluentd/cmake/opentelemetry-cpp.cmake]: Yeah lots of them :) Removed now. Thanks

File: exporters/fluentd/CMakeLists.txt (hunk 3 above)
Comment (lalitb) [reply exporters/fluentd/CMakeLists.txt]: No, was using it earlier and then forgot to remove it. Fixed now.

**Pull request merged**